share/picongpu/unit/metadata/metadataDescription.cpp

```diff
@@ -0,0 +1,328 @@
+{
+    MetadataAggregator metadataAggregator;
+
+    SECTION("can add RT info for simple object")
+    {
+        SomethingWithRTInfo obj{42};
+        auto expected = json::object();
+        expected["info"] = obj.info;
+
+        addMetadataOf(obj);
+        CHECK(metadataAggregator.metadata == expected);
+    }
+
+    SECTION("can handle multiple pieces of information")
+    {
+        SomethingWithMoreRTInfo obj{42, 'j'};
+        auto expected = json::object();
+        expected["moreInfo"] = obj.moreInfo;
+        expected["character"] = obj.c;
+
+        addMetadataOf(obj);
+        CHECK(metadataAggregator.metadata == expected);
+    }
+
+    SECTION("metadata can be customised via trait")
+    {
+        SomethingWithCustomRTInfo obj{42};
+        auto expected = json::object();
+        expected["info"] = obj.moreInfo;
+        REQUIRE(obj.metadata() != expected); // make sure we test something non-trivial here
+
+        addMetadataOf(obj);
+        CHECK(metadataAggregator.metadata == expected);
+    }
+
+    SECTION("merges two non-overlapping sets of metadata")
+    {
+        SomethingWithRTInfo obj{42};
+        SomethingWithMoreRTInfo obj2{42, 'j'};
+        auto expected = json::object();
+        expected["info"] = obj.info;
+        expected["moreInfo"] = obj2.moreInfo;
+        expected["character"] = obj2.c;
+
+        addMetadataOf(obj);
+        addMetadataOf(obj2);
+        CHECK(metadataAggregator.metadata == expected);
+    }
+
+    SECTION("overlapping metadata overwrites (test for confidence and documentation)")
+    {
+        // actually in the documentation we explicitly don't specify an order, so this is technically undefined
+        // behaviour
+        SomethingWithRTInfo obj{42};
+        SomethingWithCustomRTInfo obj2{41, 'j'}; // provides same entry as previous one
+        auto expected = json::object();
+        expected["info"] = obj2.moreInfo;
+
+        addMetadataOf(obj);
+        addMetadataOf(obj2);
+        CHECK(metadataAggregator.metadata == expected);
+    }
+
+    SECTION("customisation can extract private information from `.metadata()`")
+    {
+        int privateInfo = 42;
+        SomethingWithPrivateInfo obj{privateInfo};
+        auto expected = json::object();
+        expected["privateInfo"] = privateInfo;
+        expected["customisedInfo"] = "Some customised string.";
+        REQUIRE(obj.metadata() != expected); // make sure we test something non-trivial here
+
+        addMetadataOf(obj);
+        CHECK(metadataAggregator.metadata == expected);
+    }
+
+    SECTION("customisation can extract private information as a friend")
+    {
+        int privateInfo = 42;
+        SomethingWithoutUsefulMetadata obj{privateInfo};
+        auto expected = json::object();
+        expected["privateInfo"] = privateInfo;
+        expected["customisedInfo"] = "Some other customised string.";
+        REQUIRE(obj.metadata() != expected); // make sure we test something non-trivial here
```

Comment:
please comment above the line. Adding comments behind the line ends up sometime ins strange unreadable comments if the line or indention is changing.

Sometimes this happen
```C++
The line become longer than in the initial PR; REQUIRE(obj.metadata() != expected); 
// make sure we test something non-trivial here
```